File: board-monolithic-service/board-service/board-core/src/main/java/com/yellowsunn/simpleboard/boardservice/domain/comment/Comment.java
package com.yellowsunn.simpleboard.boardservice.domain.comment;

import com.yellowsunn.simpleboard.boardservice.domain.BaseCreatedTimeEntity;
import com.yellowsunn.simpleboard.boardservice.domain.post.Posts;
import com.yellowsunn.simpleboard.userservice.domain.user.User;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@Entity
public class Comment extends BaseCreatedTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "comment_id")
    private Long id;

    @Column(length = 3000, nullable = false)
    private String content;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_id")
    private Comment parent;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id", nullable = false)
    private Posts post;

    @Builder
    public Comment(String content, Comment parent, User user, Posts post) {
        this.content = content;
        this.parent = parent;
        this.user = user;
        this.post = post;

        if (this.parent == null) {
            this.parent = this;
        }
    }
}

File: board-service/libs/board-core/src/main/kotlin/com/yellowsunn/boardservice/domain/command/comment/Comment.kt
package com.yellowsunn.boardservice.domain.command.comment

import com.yellowsunn.boardservice.domain.command.BaseTimeEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id

@Entity
class Comment(
    val articleId: Long,
    val userId: Long,
    content: String?,
    parentCommentId: Long? = null,
    parentUserId: Long? = null,
    imageUrl: String? = null,
) : BaseTimeEntity() {
    @Id
    @Column(name = "comment_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0L

    var content: String? = content
        private set

    var parentCommentId: Long? = parentCommentId
        private set

    var parentUserId: Long? = parentCommentId
        private set

    var imageUrl: String? = imageUrl
        private set

    var likeCount: Long = 0L
        private set

    fun isReply(): Boolean {
        return parentCommentId != null
    }
}
